Feat/tabs

momolarson opened the pull request.

momolarson committed:
trying to add bends

momolarson committed:
fixed lint errors

momolarson committed:
made bends array, check for release

momolarson committed:
added proper phrase text

momolarson committed:
package.json changes

momolarson committed:
fixed bends and added vibrato

fredmeister77 committed:
Enable print-object attribute for part names (#808)

* feat(label): Add support for no-print on part names
Don't render part name if no-print attribute = no

* Add test file

Co-authored-by: Justin Litten <justin@jlitten-laptop.WORKGROUP>

sschmidTU commented on package.json:
```diff
@@ -62,11 +62,11 @@
     "svg2pdf.js": "^1.5.0",
     "typescript-collections": "^1.3.3",
-    "vexflow": "3.0.4"
+    "vexflow": "^3.0.4"
```
using the caret for the Vexflow version is not a good idea. Vexflow ^3.0.4 means any Vexflow version at or above 3.0.4, below 4.0 (compatible with 3.0.4), arbitrarily the most recent one when you last performed npm install.
There are often breaking changes in Vexflow updates, so each version should be tested and updated to precisely and carefully, without a caret.

sschmidTU replied:
by the way, we are using 1.2.93 right now, so the pull request shouldn't change this. It would be nice if you could merge opensheetmusicdisplay/develop. (not the develop in your fork)

sschmidTU commented on package.json:
```diff
@@ -62,11 +62,11 @@
     "@types/vexflow": "^1.2.33",
     "jspdf-yworks": "^2.1.1",
-    "jszip": "3.4.0",
+    "jszip": "^3.4.0",
```
this change should also not be a part of this pull request.
if you can simply revert these package.json changes, we can probably do a pull request straight to opensheetmusicdisplay/develop.

sschmidTU commented on src/MusicalScore/ScoreIO/VoiceGenerator.ts:
```diff
@@ -453,6 +454,17 @@ export class VoiceGenerator {
         }
+        const bendElementsArr: IXmlElement[] = technicalNode.elements("bend");
+        bendElementsArr.forEach(function (bend: IXmlElement): void {
+            console.log(bend);
```
(all) console.log should be removed.

sschmidTU reviewed (changes requested):
without the package.json changes and the one console.log, this is looking very good, solid code, so after these small changes i'd tend to quickly test and merge this to opensheetmusicdisplay/develop.

Commit:
merge PR#1 by momolarson (tab bend/release/vibrato)

merge https://github.com/momolarson/opensheetmusicdisplay/pull/1
resolved merge conflicts

reverted package.json changes

removed console.log in VoiceEntry.ts

Commit:
comment

sschmidTU committed:
Merge pull request #811 from opensheetmusicdisplay/feat/tabs/bend-release

fix branch name

Commit:
fix scope error

momolarson committed:
fixed merge conflicts
